src/Http/Http.Abstractions/src/PathString.cs

```diff
@@ -187,9 +192,12 @@ public static PathString FromUriComponent(Uri uri)
             {
                 throw new ArgumentNullException(nameof(uri));
             }
-
-            // REVIEW: what is the exactly correct thing to do?
-            return new PathString("/" + uri.GetComponents(UriComponents.Path, UriFormat.Unescaped));
+            var uriComponent = uri.GetComponents(UriComponents.Path, UriFormat.UriEscaped);
+            Span<char> pathBuffer = uriComponent.Length <= StackAllocThreshold ? stackalloc char[uriComponent.Length + 1] : new char[uriComponent.Length + 1];
```

Comment:
```suggestion
            Span<char> pathBuffer = uriComponent.Length <= StackAllocThreshold ? stackalloc char[StackAllocThreshold] : new char[uriComponent.Length + 1];
```

It's more efficient to use a constant for the stackalloc parameter (cc @GrabYourPitchforks )

Reply:
ack